### PR TITLE
BUG: issues/148v3

### PR DESCRIPTION
**Pull request by drakenclimber**
Changes from v2 to v3:

- Addressed code review comments from @pcmoore
- Fully incorporated the _hsh_add() patch proposed by @pcmoore in issue #148.  (I hadn't updated from an earlier iteration.)
- Added 50-sim-hash_collision to tests/.gitignore

### CODE REVIEW

File: src/gen_bpf.c [hunk 1]
```diff
@@ -569,11 +572,16 @@ static int _hsh_add(struct bpf_state *state, struct bpf_blk **blk_p,
...
+	h_val_tmp[0] = hash(blk->blks, _BLK_MSZE(blk));
+	h_val_tmp[1] = hash(&blk->acc_start, sizeof(blk->acc_start));
+	h_val_tmp[2] = hash(&blk->acc_end, sizeof(blk->acc_end));
+	h_val = hash(h_val_tmp, sizeof(h_val_tmp));
 	blk->hash = h_val;
 	blk->flag_hash = true;
 	blk->node = NULL;
 	h_new->blk = blk;
+	h_new->acc_start = blk->acc_start;
+	h_new->acc_end = blk->acc_end;
```
Comment (pcmoore) [inline src/gen_bpf.c]: I actually liked what you were doing in your patch where you simply reused the accumulator state kept inside the `blk` variable.  I'd vote for changing this to use that approach, or was there a problem with that?

**Comment by pcmoore** [comment]
Other than the small nit above, this looks good to me - thanks @drakenclimber.

File: src/gen_bpf.c (hunk 1 above)
Comment (drakenclimber) [reply src/gen_bpf.c]: I don't believe there were any issues with the approach I had taken, but I admit I would like to get more testing miles on it.

I had forgotten to push your latest patch to my git repo and was simply running with my intermediate one.  I'll grab the reused accumulator changes from my patch and merge them with this one.  I'll try to log some more testing miles as well.

Stay tuned...

Comment (pcmoore) [reply src/gen_bpf.c]: Great, thanks!

**Pull request closed without merging**